src/core/proto/rule_table_mgr.cpp

```diff
@@ -101,71 +101,74 @@ void rule_table_mgr::update_tbl(nl_data_t data_type)
+    // Cast the generic nl_object to a specific route or rule object
+    struct rtnl_rule *rule = reinterpret_cast<struct rtnl_rule *>(nl_obj);
 
-    val.set_family(rt_msg->rtm_family);
-    val.set_protocol(rt_msg->rtm_protocol);
-    val.set_scope(rt_msg->rtm_scope);
-    val.set_type(rt_msg->rtm_type);
-    val.set_tos(rt_msg->rtm_tos);
-    val.set_table_id(rt_msg->rtm_table);
+    // Set rule properties in p_val using libnl getters
+    uint8_t protocol = 0;
+    err = rtnl_rule_get_protocol(rule, &protocol);
+    if (err < 0) {
+        throw_xlio_exception("Failed to get rule protocol");
+    }
 
-    len = RTM_PAYLOAD(nl_header);
-    rt_attribute = (struct rtattr *)RTM_RTA(rt_msg);
+    val.set_family(rtnl_rule_get_family(rule));
+    val.set_protocol(protocol);
+    val.set_tos(rtnl_rule_get_dsfield(rule));
+    val.set_table_id(rtnl_rule_get_table(rule));
+
+    parse_attr(rule, val);
 
-    for (; RTA_OK(rt_attribute, len); rt_attribute = RTA_NEXT(rt_attribute, len)) {
-        parse_attr(rt_attribute, val);
-    }
     val.set_state(true);
 
     rule_table_t &table = val.get_family() == AF_INET ? m_table_in4 : m_table_in6;
     table.push_back(val);
 }
 
 // Parse received rule attribute for given rule.
-// Parameters:
-//		rt_attribute	: object that contain rule attribute.
-//		p_val			: custom object that contain parsed rule data.
-void rule_table_mgr::parse_attr(struct rtattr *rt_attribute, rule_val &val)
+void rule_table_mgr::parse_attr(struct rtnl_rule *rule, rule_val &val)
 {
-    switch (rt_attribute->rta_type) {
-    case FRA_PRIORITY:
-        val.set_priority(*(uint32_t *)RTA_DATA(rt_attribute));
-        break;
-    case FRA_DST:
-        val.set_dst_addr(ip_address((void *)RTA_DATA(rt_attribute), val.get_family()));
-        break;
-    case FRA_SRC:
-        val.set_src_addr(ip_address((void *)RTA_DATA(rt_attribute), val.get_family()));
-        break;
-    case FRA_IFNAME:
-        val.set_iif_name((char *)RTA_DATA(rt_attribute));
-        break;
-    case FRA_TABLE:
-        val.set_table_id(*(uint32_t *)RTA_DATA(rt_attribute));
-        break;
+    // FRA_PRIORITY: Rule Priority
+    uint32_t priority = rtnl_rule_get_prio(rule);
+    if (priority) {
+        val.set_priority(priority);
+    }
+
+    // FRA_DST: Destination Address
+    struct nl_addr *dst = rtnl_rule_get_dst(rule);
+    if (dst) {
+        val.set_dst_addr(ip_address(nl_addr_get_binary_addr(dst), val.get_family()));
+    }
+
+    // FRA_SRC: Source Address
+    struct nl_addr *src = rtnl_rule_get_src(rule);
+    if (src) {
+        val.set_src_addr(ip_address(nl_addr_get_binary_addr(src), val.get_family()));
+    }
+
+    // FRA_IFNAME: Input Interface Name
+    char *iif_name = rtnl_rule_get_iif(rule);
+    if (iif_name) {
+        val.set_iif_name(iif_name);
+    }
+
+    // FRA_TABLE: Table ID
+    uint32_t table_id = rtnl_rule_get_table(rule);
+    if (table_id) {
+        val.set_table_id(table_id);
+    }
+
 #if DEFINED_FRA_OIFNAME
```

Comment:
Remove this ifdef and remove its definition in configure.ac. All libnl-3 versions provide rtnl_rule_get_oif().